src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,182 @@
+        String[] storageStrArr = task.toStorageStringArr();
+        String joinedStorageStr = String.join(DELIMITER, storageStrArr);
+        this.fileBuffer.add(joinedStorageStr);
+    }
+
+
+    /**
+     * Writes file buffer to file.
+     */
+    public void writeToFile() {
+        try {
+            if (fileBuffer.size() > 0) {
+                if (!fileExists()) {
+                    // creates the file if not found
+                    boolean isFileCreated = this.createFile();
+                }
+
+                FileWriter fWriter = new FileWriter(storageFile);
+                BufferedWriter writer = new BufferedWriter(fWriter);
+
+                for (String bufferedLine : this.fileBuffer) {
+                    writer.append(bufferedLine).append("\n");
+                }
+
+                writer.close();
+            }
+            // else don't need to write
+        } catch (IOException exception) {
+            System.out.println("Error writing to file!");
+            exception.printStackTrace();
+        }
+    }
+
+
+    /**
+     * Parses line input to an array.
+     *
+     * @param inputString input string.
+     * @return parsed array, Array[0] => itemString, Array[1] => isDone.
+     */
+    private static String[] parseLine(String inputString) {
+        String[] splitStr = inputString.split(DELIMITER);
+
+        String taskType = splitStr[0];
+        String status = splitStr[1];
+        String taskStr = splitStr[2];
+
+        StringBuilder parsedStringBuilder = new StringBuilder();
+
+        switch (taskType) {
+        case ("T"):
+            parsedStringBuilder.append("todo ").append(taskStr);
+            break;
+        case ("E"):
+            parsedStringBuilder.append("event ").append(taskStr).append(" /at ").append(splitStr[3]);
+            break;
+        case ("D"):
+            parsedStringBuilder.append("deadline ").append(taskStr).append(" /by ").append(splitStr[3]);
+            break;
+        }
+
+        return new String[]{parsedStringBuilder.toString(), status};
+    }
+
+
+    /**
+     * Reads file for saved tasks.
+     *
+     * @return 2D array of saved tasks. Each task is given in an array of size 2.
+     */
+    public String[][] readFromFile() {
+        try {
+            ArrayList<String[]> parsedLines = new ArrayList<>();
+
+            BufferedReader reader = new BufferedReader(new FileReader(this.storageFile));
+
+            String currLine = reader.readLine();
+            while (currLine != null) {
+                parsedLines.add(parseLine(currLine));
+                currLine = reader.readLine();
+            }
+
+            return parsedLines.toArray(new String[0][0]);
+
+        } catch (FileNotFoundException exception) {
+            return new String[0][0];
+        } catch (IOException exception) {
+            System.out.println("Error reading file!");
+            exception.printStackTrace();
+            return new String[0][0];
+        }
+    }
+
+}
```

Comment:
Overall good formatting / use of Java coding standards for this particular script Storage.java. Good job! 👍🏻 👍🏻 👍🏻